src/main/java/pl/onsight/wypozyczalnia/validator/impl/DateValidatorImpl.java

```diff
@@ -0,0 +1,47 @@
+package pl.onsight.wypozyczalnia.validator.impl;
+
+import org.springframework.stereotype.Component;
+import pl.onsight.wypozyczalnia.DateFilter;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+@Component
+public class DateValidatorImpl implements pl.onsight.wypozyczalnia.validator.DateValidator {
+
+    @Override
+    public boolean isDateValid(String dates) throws ParseException {
+        System.out.println(dates);
+        return isDatesCorrectToParse(dates) && areDatesCorrect(dates) ;
+    }
+    private boolean areDatesCorrect(String dates) throws ParseException {
+        Date[] datesArray = DateFilter.changeStringToDate(dates);
+        return areDatesInFuture(datesArray[0], datesArray[1]) && isDateStartBeforeDateEnd(datesArray[0], datesArray[1]);
+    }
+
+    private boolean areDatesInFuture(Date start, Date end) throws ParseException {
+        SimpleDateFormat sdf = new SimpleDateFormat("dd/MM/yyyy HH:mm");
+        Date today = sdf.parse(sdf.format(new Date()).split(" ")[0]+ " 00:00");
+        return !start.before(today) && !end.before(today);
+    }
+
+    private boolean isDateStartBeforeDateEnd(Date start, Date end){
+        return !start.after(end);
```

Comment:
nie mogloby byc tutaj start.before(end)? jesli efekt bylby dobry to lepiej tak, bo nie ma zaprzeczania i jest czytelniej